### PR TITLE
22007

### PR DESCRIPTION
**Pull request by mbroz2**
https://blogs-draft-openlibertyio.mybluemix.net/blog/2022/07/05/logrecordcontext-22007.html
image

### CODE REVIEW

File: posts/2022-07-05-logrecordcontext-22007.adoc [hunk 1]
```diff
@@ -0,0 +1,336 @@
...
+  "ibm_sequence":"1645038467226_000000000001F",
+  "ext_appName":"TestApp",
+  "ext_thread":"Default Executor-thread-31"
+}
+----
+
+For more information, check out the link:{url-prefix}/docs/latest/log-trace-configuration.html#json[JSON Logging] and link:{url-prefix}/docs/latest/json-log-events-list.html[JSON Log Events list] documentation.
+
+[#logging]
+== New Logging functionality: merge stack traces into a single log event
```
Comment (lauracowen) [inline posts/2022-07-05-logrecordcontext-22007.adoc]: I don't think it needs to include "New logging functionality:" in the heading - that it's new functionality goes without saying in the context of this blog post.

But also, I don't think this section ever actually tells the reader what the benefit is of the update. I'm guessing that the benefit is that the stack traces are now more readable within the log files (or can be configured to be) because the stack traces are organised/printed/grouped as a single event.

Ideally, I'd reword that whole section as something like:

"You can now configure the stack trace in Open Liberty logs to be more readable by printing stack traces as a single log event. Previously, each line of the stack trace, typically thrown by `Throwable.printStackTrace()` calls from the application, was printed as a separate event.

Now, you can configure Open Liberty to output the emitted stack trace as a single log event. This is useful when using third-party log analysis tools, such as the Elastic Logstash Kibana (ELK) stack, because the xxxx(what? how do the tools benefit from stack trace lines being grouped together - that's not clear from this text?).

This configuration applies to the console output, console logs, messages log, and the trace logs. The following examples show the console output and the messages log when the stack traces are printed as separate events versus as a single event:"

I know rewording it might not be an option but the main point is that the benefit when using log analysis tools needs to be added.

**Review by lauracowen — COMMENTED** [review]
Would it work for the title (and opening paragraph and relevant headings) to be something like "Filter logs on application name in Open Liberty 22.00.7" - focusing on the benefit rather than how it's done?

Or if that by itself doesn't sound a big deal to the target audience, maybe could focus on the logs angle like "Improved log filtering and more readable stack traces events in Open Liberty 22.0.0.7" (not certain that's perfectly technically accurate wording, but something like that - I'm not sure "readable" is quite the right term when the benefit is when using log analysis tools, I think?).

(We don't need to cover everything in the title if it doesn't fit the space though.)

File: posts/2022-07-05-logrecordcontext-22007.adoc [hunk 2]
```diff
@@ -0,0 +1,336 @@
...
+Open Liberty 22.0.0.7 includes the addition of the application name to the LogRecordContext extension and a corresponding JSON logging field, as well as providing new logging functionality to join stack traces in logs so that they are output as a single log event. Also included in this release are several notable bug fixes and updates to our Open Liberty guides.
+
+In link:{url-about}[Open Liberty] 22.0.0.7:
+
+* <<logrecordcontext, Add application name to LogRecordContext extension and as JSON logging field>>
+* <<logging, New Logging functionality: merge stack traces into a single log event>>
+* <<bugs, Notable bug fixes>>
+* <<guides, New and updated guides since the previous release>>
+
+Along with the new features and functions added to the runtime, we’ve also made <<guides, updates to our guides>>.
```
Comment (dmuelle) [inline posts/2022-07-05-logrecordcontext-22007.adoc:28]: here and line 31 seem redundant as the guide updates and bug fixes have already been mentioned twice in the preceding lines

File: posts/2022-07-05-logrecordcontext-22007.adoc [hunk 3]
```diff
@@ -0,0 +1,336 @@
...
+  "ext_thread":"Default Executor-thread-31"
+}
+----
+
+For more information, check out the link:{url-prefix}/docs/latest/log-trace-configuration.html#json[JSON Logging] and link:{url-prefix}/docs/latest/json-log-events-list.html[JSON Log Events list] documentation.
+
+[#logging]
+== New Logging functionality: merge stack traces into a single log event
+
+When a stack trace is logged in Open Liberty, typically from `Throwable.printStackTrace()` calls from the application, you can now output the emitted stack trace as a single log event. This update is impactful if you forward your logs downstream to third-party log analysis technologies, such as the Elastic Logstash Kibana (ELK) stack. It affects the console output, console logs, messages log, and the trace logs. Prior to this update, each line of the stack trace was printed as a separate event.
```
Comment (dmuelle) [inline posts/2022-07-05-logrecordcontext-22007.adoc]: the Elastic Logstash Kibana (ELK) stack is already mentioned in the previous section- if we're going to spell it out, should do so on first mention at line 72

File: posts/2022-07-05-logrecordcontext-22007.adoc [hunk 4]
```diff
@@ -0,0 +1,336 @@
...
+
+Or take a look at our link:{url-prefix}/downloads/[Downloads page].
+
+[link=https://stackoverflow.com/tags/open-liberty]
+image::img/blog/blog_btn_stack.svg[Ask a question on Stack Overflow, align="center"]
+
+[#logrecordcontext]
+== Add application name to LogRecordContext extension and as JSON logging field
+
+Previously, Liberty logs did not contain the application name in the `LogRecordContext` extension. If you were using a log analysis tool, such as the ELK stack, you couldn't filter out application logs, since the `JSON` fields did not have a field for the application name.
```
Comment (dmuelle) [inline posts/2022-07-05-logrecordcontext-22007.adoc]: spell out Elastic Logstash Kibana (ELK) on first mention and abbreviate ELK for subsequent mentions

File: posts/2022-07-05-logrecordcontext-22007.adoc [hunk 5]
```diff
@@ -0,0 +1,336 @@
...
+[source]
+----
+WLP_LOGGING_STACK_TRACE_SINGLE_ENTRY=true
+----
+- `server.xml`: 
+[source,xml]
+----
+<logging stackTraceSingleEntry="true" />
+----
+
```
Comment (dmuelle) [inline posts/2022-07-05-logrecordcontext-22007.adoc:214]:
```suggestion
For more information, see the entry for `stackTraceSingleEntry` in link:{url-prefix}/docs/latest/log-trace-configuration.html#settings[Configuration settings by source].

```

File: posts/2022-07-05-logrecordcontext-22007.adoc [hunk 6]
```diff
@@ -0,0 +1,336 @@
...
+  - ...
+  type: object
+  properties:
+    asset_id:       <<----- name of the property in the "properties" section is correct
+      type: string
+      readOnly: true
+    ...
+----
++
+The property is generated correctly in the properties section however it was not generated correctly in the required section. The expected name should be the name used in the `@Schema` annotation, i.e., `asset_id`. This issue has now been resolved so that the correct property name for the `@Schema` annotation is generated.
```
Comment (dmuelle) [inline posts/2022-07-05-logrecordcontext-22007.adoc]:
```suggestion
Although the property is generated correctly in the properties section, it was not generated correctly in the required section. The expected name should be the name used in the `@Schema` annotation, i.e., `asset_id`. This issue has now been resolved so that the correct property name for the `@Schema` annotation is generated.
```

File: posts/2022-07-05-logrecordcontext-22007.adoc [hunk 7]
```diff
@@ -0,0 +1,336 @@
...
+
+* link:https://github.com/OpenLiberty/open-liberty/issues/21148[Transactions summary trace is missing]
++
+We discovered that the `TransactionSummary` trace group which is used to provide ultra minimal trace for transaction creation and setRollbackOnly no longer works.
```
Comment (dmuelle) [inline posts/2022-07-05-logrecordcontext-22007.adoc]:
```suggestion
We discovered that the `TransactionSummary` trace group was no longer working. This trace group provides ultra minimal trace for transaction creation and setRollbackOnly .
```

**Review by dmuelle — CHANGES_REQUESTED** [review]
ID review- made a few suggestions. Agree with Laura that the title and headings could be more descriptive of the value users get rather than the changes we made to provide that value. In the first section- value to the user is most likely the ability to filter JSON logs on the app name (i think?). If so we should lead with that, rather than focusing on the LogRecordContext extension, which is an internal detail that doesn't make the use value evident.

File: posts/2022-07-05-logrecordcontext-22007.adoc [hunk 8]
```diff
@@ -0,0 +1,336 @@
...
+FROM open-liberty
+----
+
+Or take a look at our link:{url-prefix}/downloads/[Downloads page].
+
+[link=https://stackoverflow.com/tags/open-liberty]
+image::img/blog/blog_btn_stack.svg[Ask a question on Stack Overflow, align="center"]
+
+[#logrecordcontext]
+== Add application name to LogRecordContext extension and as JSON logging field
```
Comment (dmuelle) [inline posts/2022-07-05-logrecordcontext-22007.adoc]: maybe
```suggestion
== Filter JSON logs on the application name
```

File: posts/2022-07-05-logrecordcontext-22007.adoc [hunk 9]
```diff
@@ -0,0 +1,336 @@
...
+//Blank line here is necessary before starting the body of the post.
+
+Open Liberty 22.0.0.7 includes the addition of the application name to the LogRecordContext extension and a corresponding JSON logging field, as well as providing new logging functionality to join stack traces in logs so that they are output as a single log event. Also included in this release are several notable bug fixes and updates to our Open Liberty guides.
+
+In link:{url-about}[Open Liberty] 22.0.0.7:
+
+* <<logrecordcontext, Add application name to LogRecordContext extension and as JSON logging field>>
+* <<logging, New Logging functionality: merge stack traces into a single log event>>
+* <<bugs, Notable bug fixes>>
+* <<guides, New and updated guides since the previous release>>
```
Comment (mbroz2) [inline posts/2022-07-05-logrecordcontext-22007.adoc]:
```suggestion
```

File: posts/2022-07-05-logrecordcontext-22007.adoc [hunk 10]
```diff
@@ -0,0 +1,336 @@
...
+  "ext_thread":"Default Executor-thread-31"
+}
+----
+
+For more information, check out the link:{url-prefix}/docs/latest/log-trace-configuration.html#json[JSON Logging] and link:{url-prefix}/docs/latest/json-log-events-list.html[JSON Log Events list] documentation.
+
+[#logging]
+== New Logging functionality: merge stack traces into a single log event
+
+When a stack trace is logged in Open Liberty, typically from `Throwable.printStackTrace()` calls from the application, you can now output the emitted stack trace as a single log event. This update is impactful if you forward your logs downstream to third-party log analysis technologies, such as the Elastic Logstash Kibana (ELK) stack. It affects the console output, console logs, messages log, and the trace logs. Prior to this update, each line of the stack trace was printed as a separate event.
```
Comment (mbroz2) [inline posts/2022-07-05-logrecordcontext-22007.adoc]:
```suggestion
When a stack trace is logged in Open Liberty, typically from `Throwable.printStackTrace()` calls from the application, you can now output the emitted stack trace as a single log event. This update is impactful if you forward your logs downstream to third-party log analysis technologies, such as the ELK stack. It affects the console output, console logs, messages log, and the trace logs. Prior to this update, each line of the stack trace was printed as a separate event.
```

File: posts/2022-07-05-logrecordcontext-22007.adoc [hunk 11]
```diff
@@ -0,0 +1,336 @@
...
+
+Or take a look at our link:{url-prefix}/downloads/[Downloads page].
+
+[link=https://stackoverflow.com/tags/open-liberty]
+image::img/blog/blog_btn_stack.svg[Ask a question on Stack Overflow, align="center"]
+
+[#logrecordcontext]
+== Add application name to LogRecordContext extension and as JSON logging field
+
+Previously, Liberty logs did not contain the application name in the `LogRecordContext` extension. If you were using a log analysis tool, such as the ELK stack, you couldn't filter out application logs, since the `JSON` fields did not have a field for the application name.
```
Comment (mbroz2) [inline posts/2022-07-05-logrecordcontext-22007.adoc]:
```suggestion
Previously, Liberty logs did not contain the application name in the `LogRecordContext` extension. If you were using a log analysis tool, such as the Elastic Logstash Kibana (ELK) stack, you couldn't filter out application logs, since the `JSON` fields did not have a field for the application name. 
```

File: posts/2022-07-05-logrecordcontext-22007.adoc [hunk 12]
```diff
@@ -0,0 +1,336 @@
...
+  "ext_thread":"Default Executor-thread-31"
+}
+----
+
+For more information, check out the link:{url-prefix}/docs/latest/log-trace-configuration.html#json[JSON Logging] and link:{url-prefix}/docs/latest/json-log-events-list.html[JSON Log Events list] documentation.
+
+[#logging]
+== New Logging functionality: merge stack traces into a single log event
+
+When a stack trace is logged in Open Liberty, typically from `Throwable.printStackTrace()` calls from the application, you can now output the emitted stack trace as a single log event. This update is impactful if you forward your logs downstream to third-party log analysis technologies, such as the ELK stack. It affects the console output, console logs, messages log, and the trace logs. Prior to this update, each line of the stack trace was printed as a separate event.
```
Comment (mbroz2) [inline posts/2022-07-05-logrecordcontext-22007.adoc]:
```suggestion
[#logging]
== Merge stack traces into a single log event

You can now configure the stack trace in Open Liberty logs to be more readable by printing stack traces as a single log event. Previously, each line of the stack trace, typically thrown by Throwable.printStackTrace() calls from the application, was printed as a separate event.

Now, you can configure Open Liberty to output the emitted stack trace as a single log event. This is useful when using third-party log analysis tools, such as the Elastic Logstash Kibana (ELK) stack, so that each stack trace is treated as a single event instead of multiple.

This configuration applies to the console output, console logs, messages log, and the trace logs. The following examples show the console output and the messages log when the stack traces are printed as separate events versus as a single event:

```

File: posts/2022-07-05-logrecordcontext-22007.adoc [hunk 13]
```diff
@@ -0,0 +1,336 @@
...
+
+*Examples:*
```
Comment (mbroz2) [inline posts/2022-07-05-logrecordcontext-22007.adoc]:
```suggestion
```

File: posts/2022-07-05-logrecordcontext-22007.adoc [hunk 14]
```diff
@@ -0,0 +1,339 @@
+---
+layout: post
+title: "Enhancements to LogRecordContext extension and improved stack traces in Open Liberty 22.0.0.7"
```
Comment (mbroz2) [inline posts/2022-07-05-logrecordcontext-22007.adoc]:
```suggestion
title: "Improved log filtering and more consumable stack traces in Open Liberty 22.0.0.7"
```

File: posts/2022-07-05-logrecordcontext-22007.adoc [hunk 15]
```diff
@@ -0,0 +1,339 @@
...
+# Do NOT change the categories section
+categories: blog
+author_picture: https://avatars3.githubusercontent.com/ryan-storey
+author_github: https://github.com/ryan-storey
+seo-title: Enhancements to LogRecordContext extension and improved stack traces in Open Liberty 22.0.0.7 - OpenLiberty.io
```
Comment (mbroz2) [inline posts/2022-07-05-logrecordcontext-22007.adoc]:
```suggestion
seo-title: Improved log filtering and more consumable stack traces in Open Liberty 22.0.0.7 - OpenLiberty.io
```

File: posts/2022-07-05-logrecordcontext-22007.adoc [hunk 16]
```diff
@@ -0,0 +1,339 @@
...
+seo-description: Open Liberty 22.0.0.7 includes the addition of the application name to the LogRecordContext extension and a corresponding JSON logging field, as well as providing new logging functionality to join stack traces in logs so that they are output as a single log event. Also included in this release are several notable bug fixes and updates to our Open Liberty guides.
+blog_description: Open Liberty 22.0.0.7 includes the addition of the application name to the LogRecordContext extension and a corresponding JSON logging field, as well as providing new logging functionality to join stack traces in logs so that they are output as a single log event. Also included in this release are several notable bug fixes and updates to our Open Liberty guides.
+open-graph-image: https://openliberty.io/img/twitter_card.jpg
+---
+= Enhancements to LogRecordContext extension and improved stack traces in Open Liberty 22.0.0.7
```
Comment (mbroz2) [inline posts/2022-07-05-logrecordcontext-22007.adoc]:
```suggestion
= Improved log filtering and more consumable stack traces in Open Liberty 22.0.0.7
```

File: posts/2022-07-05-logrecordcontext-22007.adoc [hunk 17]
```diff
@@ -0,0 +1,339 @@
...
+:url-prefix:
+:url-about: /
+//Blank line here is necessary before starting the body of the post.
+
+Open Liberty 22.0.0.7 includes the addition of the application name to the LogRecordContext extension and a corresponding JSON logging field, as well as providing new logging functionality to join stack traces in logs so that they are output as a single log event. Also included in this release are several notable bug fixes and updates to our Open Liberty guides.
+
+In link:{url-about}[Open Liberty] 22.0.0.7:
+
+* <<logrecordcontext, Add application name to LogRecordContext extension and as JSON logging field>>
+* <<logging, New Logging functionality: merge stack traces into a single log event>>
```
Comment (mbroz2) [inline posts/2022-07-05-logrecordcontext-22007.adoc]:
```suggestion
* <<logging, Merge stack traces into a single log event>>
```

File: posts/2022-07-05-logrecordcontext-22007.adoc [hunk 18]
```diff
@@ -0,0 +1,339 @@
...
+open-graph-image: https://openliberty.io/img/twitter_card.jpg
+---
+= Enhancements to LogRecordContext extension and improved stack traces in Open Liberty 22.0.0.7
+Ryan Storey <https://github.com/ryan-storey>
+:imagesdir: /
+:url-prefix:
+:url-about: /
+//Blank line here is necessary before starting the body of the post.
+
+Open Liberty 22.0.0.7 includes the addition of the application name to the LogRecordContext extension and a corresponding JSON logging field, as well as providing new logging functionality to join stack traces in logs so that they are output as a single log event. Also included in this release are several notable bug fixes and updates to our Open Liberty guides.
```
Comment (mbroz2) [inline posts/2022-07-05-logrecordcontext-22007.adoc]:
```suggestion
Open Liberty 22.0.0.7 includes the addition of the application name to the LogRecordContext extension and a corresponding JSON logging field, as well as providing new logging functionality to join stack traces in logs so that they are output as a single log event and easier to consume. Also included in this release are several notable bug fixes and updates to our Open Liberty guides.
```

File: posts/2022-07-05-logrecordcontext-22007.adoc [hunk 19]
```diff
@@ -0,0 +1,339 @@
+---
+layout: post
+title: "Enhancements to LogRecordContext extension and improved stack traces in Open Liberty 22.0.0.7"
+# Do NOT change the categories section
+categories: blog
+author_picture: https://avatars3.githubusercontent.com/ryan-storey
+author_github: https://github.com/ryan-storey
+seo-title: Enhancements to LogRecordContext extension and improved stack traces in Open Liberty 22.0.0.7 - OpenLiberty.io
+seo-description: Open Liberty 22.0.0.7 includes the addition of the application name to the LogRecordContext extension and a corresponding JSON logging field, as well as providing new logging functionality to join stack traces in logs so that they are output as a single log event. Also included in this release are several notable bug fixes and updates to our Open Liberty guides.
+blog_description: Open Liberty 22.0.0.7 includes the addition of the application name to the LogRecordContext extension and a corresponding JSON logging field, as well as providing new logging functionality to join stack traces in logs so that they are output as a single log event. Also included in this release are several notable bug fixes and updates to our Open Liberty guides.
```
Comment (mbroz2) [inline posts/2022-07-05-logrecordcontext-22007.adoc]:
```suggestion
blog_description: Open Liberty 22.0.0.7 includes the addition of the application name to the LogRecordContext extension and a corresponding JSON logging field, as well as providing new logging functionality to join stack traces in logs so that they are output as a single log event and easier to consume. Also included in this release are several notable bug fixes and updates to our Open Liberty guides.
```

File: posts/2022-07-05-logrecordcontext-22007.adoc [hunk 20]
```diff
@@ -0,0 +1,339 @@
+---
+layout: post
+title: "Enhancements to LogRecordContext extension and improved stack traces in Open Liberty 22.0.0.7"
+# Do NOT change the categories section
+categories: blog
+author_picture: https://avatars3.githubusercontent.com/ryan-storey
+author_github: https://github.com/ryan-storey
+seo-title: Enhancements to LogRecordContext extension and improved stack traces in Open Liberty 22.0.0.7 - OpenLiberty.io
+seo-description: Open Liberty 22.0.0.7 includes the addition of the application name to the LogRecordContext extension and a corresponding JSON logging field, as well as providing new logging functionality to join stack traces in logs so that they are output as a single log event. Also included in this release are several notable bug fixes and updates to our Open Liberty guides.
```
Comment (mbroz2) [inline posts/2022-07-05-logrecordcontext-22007.adoc]:
```suggestion
seo-description: Open Liberty 22.0.0.7 includes the addition of the application name to the LogRecordContext extension and a corresponding JSON logging field, as well as providing new logging functionality to join stack traces in logs so that they are output as a single log event and easier to consume. Also included in this release are several notable bug fixes and updates to our Open Liberty guides.
```

File: posts/2022-07-05-logrecordcontext-22007.adoc (hunk 3 above)
Comment (lauracowen) [reply posts/2022-07-05-logrecordcontext-22007.adoc]: ELK is mentioned but it doesn't say why this change is "impactful"/beneficial for using ELK.
